### PR TITLE
Add workbench created and updated telemetry events

### LINKED ISSUE #807: [Bug]: Failed to Fire `Notebook Server Started` Telemetry Event
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

When we start a workbench in the projects UI, we need to fire the started event for the notebook so telemetry can have a similar feature to the way the Jupyter tile starts notebooks. 

### Expected Behavior

Add [the trigger code](https://github.com/opendatahub-io/odh-dashboard/blob/main/frontend/src/pages/notebookController/screens/server/SpawnerPage.tsx#L231-L237) to the workbench submit code before it redirects.

### Steps To Reproduce

1. Submit a workbench
2. Check to see if we fired the telemetry event

### Workaround (if any)

_No response_

### What browsers are you seeing the problem on?

_No response_

### Open Data Hub Version

```yml
2.5.0
```

### Anything else

_No response_

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
## Description

Closes #807 

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [ ] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx [hunk 1]
```diff
@@ -49,7 +50,13 @@ const SpawnerFooter: React.FC<SpawnerFooterProps> = ({
     !checkRequiredFieldsForNotebookStart(startNotebookData, storageData, envVariables);
   const { username } = useUser();
 
-  const redirect = () => {
+  const afterStart = (type: 'created' | 'updated') => {
+    const { gpus, notebookSize, image } = startNotebookData;
+    fireTrackingEvent(`Workbench ${type}`, {
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx:55]: Anyway for us to test this locally? I'm not getting events fired

Comment (DaoDaoNoCode) [reply frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx:55]: hmm, I think we can make this event print to the console if we run DEV mode. WDYT?

Comment (andrewballantyne) [reply frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx:55]: Hmmm, yeah I suppose that's not a bad idea.

Something like: `Telmetry event triggered: ${event info}`

File: frontend/src/utilities/segmentIOUtils.tsx [hunk 2]
```diff
@@ -1,20 +1,26 @@
...
-      case 'identify':
-        window.analytics.identify(properties?.anonymousID, { clusterID });
-        break;
-      case 'page':
-        window.analytics.page(undefined, { clusterID });
-        break;
-      default:
-        window.analytics.track(eventType, { ...properties, clusterID });
+  if (DEV_MODE) {
+    console.log(`Telemetry event triggered: ${eventType} - ${stringify(properties)}`);
```
Comment (andrewballantyne) [inline frontend/src/utilities/segmentIOUtils.tsx:11]: Why are you using yaml stringify? 😕

Comment (DaoDaoNoCode) [reply frontend/src/utilities/segmentIOUtils.tsx:11]: hmm, i don't know if there is a better way to print objects

Comment (andrewballantyne) [reply frontend/src/utilities/segmentIOUtils.tsx:11]: `JSON.stringify(obj, null, 2)` will print the object in a pretty fashion with two spaces for each indent. But honestly you probably don't need to consume multiple lines. Just do `JSON.stringify(obj)` and leave it at that

**Review by andrewballantyne — CHANGES_REQUESTED** [review]
`Telemetry event triggered: page - undefined`

Can you do a check to see if you have properties first?

File: frontend/src/utilities/segmentIOUtils.tsx (hunk 2 above)
Comment (andrewballantyne) [inline frontend/src/utilities/segmentIOUtils.tsx:11]:
```suggestion
    console.log(`Telemetry event triggered: ${eventType}${properties ? ` - ${JSON.stringify(properties)}` : ''}`);
```

Comment (DaoDaoNoCode) [reply frontend/src/utilities/segmentIOUtils.tsx]: good suggestion, updated

**Review by andrewballantyne — APPROVED** [review]

File: frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx [hunk 3]
```diff
@@ -49,7 +50,13 @@ const SpawnerFooter: React.FC<SpawnerFooterProps> = ({
...
+      GPU: gpus,
+      lastSelectedSize: notebookSize.name,
+      lastSelectedImage: `${image.imageVersion?.from.name}`,
+    });
```
Comment (andrewballantyne) [inline frontend/src/pages/projects/screens/spawner/SpawnerFooter.tsx:59]: Can we add the project name?

**Review by andrewballantyne — APPROVED** [review]

**Pull request merged**